src/translation/translate.js

```diff
@@ -372,19 +376,27 @@ Zotero.Translate.Sandbox = {
 				return translation.getTranslators();
 			};
 			
-			var doneHandlerSet = false;
 			safeTranslator.translate = async function () {
 				translate.incrementAsyncProcesses("safeTranslator#translate()");
 				setDefaultHandlers(translate, translation);
-				if(!doneHandlerSet) {
-					doneHandlerSet = true;
-					translation.setHandler("done", function() { translate.decrementAsyncProcesses("safeTranslator#translate()") });
-				}
 				if(!errorHandlerSet) {
 					errorHandlerSet = true;
 					translation.setHandler("error", function(obj, error) { translate.complete(false, error) });
 				}
-				return translation.translate(false);
+				return translation.translate(false)
+					.then((returnValue) => {
+						if (Array.isArray(returnValue)) {
+							returnValue = returnValue.map((item) => {
+								item = item.wrappedJSObject ? item.wrappedJSObject : item;
+								if (item && typeof item.complete === 'function') {
+									prepareItemForParent(item);
+								}
+								return item;
+							});
+						}
+						return returnValue;
+					})
+					.finally(() => translate.decrementAsyncProcesses("safeTranslator#translate()"));
```

Comment:
Let's use `try {} finally {}` with an `let returnValue = await translation.translate()` instead of `then`.